parsers/BG.py

```diff
@@ -1,72 +1,109 @@
-
-    response = res.json()
-
-    logger.debug(f"Raw generation breakdown: {response}")
-
-    datapoints = []
-    for row in response:
-        for k in TYPE_MAPPING:
-            if row[0].startswith(k):
-                datapoints.append((TYPE_MAPPING[k], row[1]))
-                break
+def _fetch_all_production_events(
+    zone_key: ZoneKey,
+    session: Session,
+    target_datetime: datetime | None,
+    logger: Logger,
+) -> list[ProductionBreakdown]:
+    """Fetches all production events from the backend source API."""
+    if target_datetime is not None:
+        raise ParserException(
+            PARSER, "This parser is not yet able to parse historical data", zone_key
+        )
+
+    time = datetime.now(timezone.utc)
+    response = session.get(SOURCE_API_URL)
+    if not response.status_code == 200:
+        raise ParserException(
+            PARSER,
+            f"Exception when fetching production error code: {response.status_code}: {response.text}",
+            zone_key,
+        )
+
+    response_payload = response.json()
+    logger.debug(f"Raw generation breakdown: {response_payload}")
 
     production = {}
-    for k, v in datapoints:
-        production[k] = production.get(k, 0.0) + v
-
-    data = {
-        "zoneKey": zone_key,
-        "production": production,
-        "storage": {},
-        "source": "eso.bg",
-        "datetime": datetime.now(timezone.utc),
-    }
-
-    return data
+    for header, value in response_payload:
+        for production_type_bg in PRODUCTION_TYPES_BG_TO_EN:
+            if header.startswith(production_type_bg):
+                production_type_en = PRODUCTION_TYPES_BG_TO_EN[production_type_bg]
+                production[production_type_en] = (
+                    production.get(production_type_en, 0.0) + value
+                )
+
+    event = ProductionBreakdown.create(
+        logger=logger,
+        zoneKey=zone_key,
+        datetime=time,
+        source=SOURCE,
+        production=ProductionMix(
+            biomass=production.get("biomass"),
+            coal=production.get("coal"),
+            gas=production.get("gas"),
+            geothermal=production.get("geothermal"),
+            hydro=production.get("hydro"),
+            nuclear=production.get("nuclear"),
+            oil=production.get("oil"),
+            solar=production.get("solar"),
+            unknown=production.get("unknown"),
+            wind=production.get("wind"),
+        ),
+        storage=None,
+    )
+    return [event]
 
 
-if __name__ == "__main__":
-    """Main method, never used by the Electricity Map backend, but handy for testing."""
-
-    print("fetch_production() ->")
-    print(fetch_production())
+def fetch_production(
+    zone_key: ZoneKey = ZoneKey("BG"),
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list[dict]:
```

Comment:
question: should I have returned here just a single event like before?

Reply:
No returning a list is the correct way to go about it, even if it just has a single value.